lego/apps/notifications/tests/test_views.py

```diff
@@ -76,3 +77,148 @@ def test_change_setting_defaults(self):
                 "channels": constants.CHANNELS,
             },
         )
+
+
+class AnnouncementViewSetTestCase(BaseAPITestCase):
+
+    fixtures = [
+        "test_abakus_groups.yaml",
+        "test_users.yaml",
+        "test_events.yaml",
+        "test_companies.yaml",
+        "test_announcements.yaml",
+    ]
+
+    def setUp(self):
+        self.url = "/api/v1/announcements/"
+
+        admin_group = AbakusGroup.objects.get(name="Webkom")
+        self.authorized_user = User.objects.get(pk=9)
+        admin_group.add_user(self.authorized_user)
+        self.unauthorized_user = User.objects.get(pk=1)
+        self.authorized_user
+
+        self.unsent_announcement = Announcement.objects.get(id=5)
+        self.unsent_not_own_announcement = Announcement.objects.get(id=4)
+
+    def test_unauthorized_create(self):
+        """
+        An unauthorized user should not be able to create an Announcement
+        """
+
+        self.client.force_login(self.unauthorized_user)
+        response = self.client.post(
+            self.url, {"message": "test_message", "groups": [2], "events": [1]}
+        )
+        self.assertEquals(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_authorized_create(self):
+        """
+        An authorized user should be able to create an announcement
+        """
+
+        self.client.force_login(self.authorized_user)
+        response = self.client.post(
+            self.url,
+            {"message": "test_message", "groups": [2], "events": [1], "fromGroup": 11},
+        )
+        self.assertEquals(response.status_code, status.HTTP_201_CREATED)
+
+    def test_authorized_create_from_invalid_group(self):
```

Comment:
I would specify this to say "nonexistent" group, so there is no confusion about what a "valid" group is.